src/Controls/src/Core/Device.cs

```diff
@@ -33,16 +31,25 @@ public static class Device
 
 		static IPlatformServices s_platformServices;
 
-		/// <include file="../../docs/Microsoft.Maui.Controls/Device.xml" path="//Member[@MemberName='SetIdiom']/Docs" />
-		[EditorBrowsable(EditorBrowsableState.Never)]
-		public static void SetIdiom(TargetIdiom value) => Idiom = value;
 		/// <include file="../../docs/Microsoft.Maui.Controls/Device.xml" path="//Member[@MemberName='Idiom']/Docs" />
-		public static TargetIdiom Idiom { get; internal set; }
-
-		//TODO: Why are there two of these? This is never used...?
-		/// <include file="../../docs/Microsoft.Maui.Controls/Device.xml" path="//Member[@MemberName='SetTargetIdiom']/Docs" />
-		[EditorBrowsable(EditorBrowsableState.Never)]
-		public static void SetTargetIdiom(TargetIdiom value) => Idiom = value;
+		public static TargetIdiom Idiom
+		{
+			get
+			{
+				var idiom = DeviceInfo.Idiom;
+				if (idiom == DeviceIdiom.Tablet)
+					return TargetIdiom.Tablet;
+				if (idiom == DeviceIdiom.Phone)
+					return TargetIdiom.Phone;
+				if (idiom == DeviceIdiom.Desktop)
+					return TargetIdiom.Desktop;
+				if (idiom == DeviceIdiom.TV)
+					return TargetIdiom.TV;
+				if (idiom == DeviceIdiom.Watch)
+					return TargetIdiom.Watch;
+				return TargetIdiom.Unsupported;
+			}
+		}
```

Comment:
This is technically not very extensible, but in reality, this API needs to be obsoleted. I'll do that in another PR because that will involve many code changes.